site/docs/skylark/language.md

```diff
@@ -37,142 +29,97 @@ def fizz_buzz(n):
-the same context, so it is safe. At the end of the evaluation, the environment
-contains an entry mapping the identifier `var` to a list `[5]`; this list is
-then frozen.
+Bazel creates `var` when `foo.bzl` loads. `var` is thus part of `foo.bzl`'s context. When `fct()` runs, it does so within the context of `foo.bzl`. After evaluation for `foo.bzl` completes, the environment contains an immutable entry, `var`, with the value `[5]`.
 
-It is possible for multiple other files to load symbols from `foo.bzl` at the
-same time. For this reason, the following code is not legal:
+When another, `bar.bzl` loads symbols from `foo.bzl`, loaded values remain immutable. For this reason, the following code in `bar.bzl` is illegal:
 
 ```python
-load(":foo.bzl", "var", "fct")
+# `bar.bzl`
+load(":foo.bzl", "var", "fct") # loads `var`, and `fct` from `./foo.bzl`
 
 var.append(6)  # runtime error, the list stored in var is frozen
 
 fct()          # runtime error, fct() attempts to modify a frozen list
 ```
 
-Evaluation contexts are also created for the analysis of each custom rule. This
-means that any values that are returned from the rule's analysis are frozen.
-Note that by the time a custom rule's analysis begins, the .bzl file in which
-it is defined has already been loaded, and so the global variables are already
-frozen.
-
+Just like the above example using `bzl` files, values returned by rules are immutable too.
 
 ## Differences between BUILD and .bzl files
 
-The goal of a BUILD file is to register targets, by making calls to rules. On
-the other hand, loading a .bzl file doesn't have any side effect: .bzl files are
-only providing definitions (constants or functions).
+`BUILD` files register targets via making calls to rules. `.bzl` files provide definitions for constants, rules, macros, and functions.
 
-[Some functions](../be/functions.html) and [native rules](
-../be/overview.html#language-specific-native-rules) are exposed to BUILD files
-as global symbols. In bzl files, they are available under the [`native` module](
+[Native functions](../be/functions.html) and [native rules](
+../be/overview.html#language-specific-native-rules) are global symbols in `BUILD` files. `bzl` files need to load them using the [`native` module](
 https://docs.bazel.build/versions/master/skylark/lib/native.html).
 
-There are two syntactic restrictions in BUILD files:
-
-* Function definitions (`def` statements) are not allowed in BUILD files.
-
-* `*args` and `**kwargs` arguments are not allowed in BUILD files; instead list
-  all the arguments explicitly.
-
+There are two syntactic restrictions in `BUILD` files: 1) declaring functions is illegal, and 2) `*args` and `**kwargs` arguments are not allowed.
 
 ## Differences with Python
 
-In addition to the mutability restrictions, there are also differences with
-Python:
-
-* Global variables cannot be reassigned.
+* Global variables are immutable.
```

Comment:
This can be misread as saying that you can't do

    a = []
    a.append(1)

at the global level, which is actually allowed. Maybe keep the current wording for this point.

Reply:
Great callout! I updated the statement to make it accurate.